apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx

```diff
@@ -31,24 +49,135 @@ const ReferenceSettings: FC = () => {
-    <Menu.Item as="a" className={clsx({ 'dropdown-active': selected }, 'menu-item')} onClick={onClick}>
-      <div className="flex items-center justify-between space-x-2">
-        <div className="flex items-center space-x-1.5">
-          <div className="text-brand">{icon}</div>
-          <div>{title}</div>
+  const getClosest = (arr: string[], val: string): RANGE_VALUES => {
+    return arr.reduce(function (prev, curr) {
+      return Math.abs(Number(curr) - Number(val)) < Math.abs(Number(prev) - Number(val)) ? curr : prev;
+    }) as RANGE_VALUES;
+  };
+
+  const handlePointerUp = (event: React.PointerEvent<HTMLInputElement>) => {
+    let closest: RANGE_VALUES = getClosest(Object.values(RANGE_VALUES), event.currentTarget.value);
+    setfollowValue(closest);
+  };
+
+  useEffect(() => {
+    switch (followValue) {
+      case MY_FOLLOWS.value:
+        setSelectedReferenceModule(ReferenceModules.DegreesOfSeparationReferenceModule);
+        setDegreesOfSeparation(1);
+        break;
+      case MY_FOLLOWERS.value:
+        setSelectedReferenceModule(ReferenceModules.FollowerOnlyReferenceModule);
+        setOnlyFollowers(true);
+        break;
+      case FRIENDS_OF_FRIENDS.value:
+        setSelectedReferenceModule(ReferenceModules.DegreesOfSeparationReferenceModule);
+        setDegreesOfSeparation(2);
+        break;
+      case EVERYONE.value:
+        setSelectedReferenceModule(ReferenceModules.FollowerOnlyReferenceModule);
+        setOnlyFollowers(false);
+        break;
+      default:
+        break;
+    }
+  }, [followValue]);
+
+  const Module: FC<ModuleProps> = ({ title, description, icon }) => (
+    <div className="flex items-center space-x-3">
+      {title === EVERYONE.title && (
+        <div className="relative flex h-20 w-20 items-center justify-center">
+          <div className="bg-brand-500 absolute h-12 w-12 rounded-full opacity-70 mix-blend-multiply blur-md filter" />
+          <div className="bg-brand-500 absolute h-8 w-8 rounded-full opacity-50 mix-blend-multiply blur-sm filter" />
+          <div className="text-brand-100 absolute">{icon}</div>
         </div>
-        {selected && <CheckCircleIcon className="w-5 text-green-500" />}
+      )}
+      {title !== EVERYONE.title && (
+        <div className="border-brand-300 dark:border-brand-400 rounded-full border-2 p-1.5">
+          <div className="border-brand-300 dark:border-brand-400 rounded-full border-2 p-2">
+            <div className="bg-brand-500 border-brand-500 rounded-full border-2 bg-opacity-30 p-2.5">
+              <div className="bg-brand-500 rounded-full p-1">
+                <div className="text-brand-100">{icon}</div>
+              </div>
+            </div>
+          </div>
+        </div>
+      )}
+      <div>
+        <div className="text-brand-500 text-lg font-semibold">{title}</div>
+        <div>{description}</div>
       </div>
-    </Menu.Item>
+    </div>
+  );
+
+  const Slider: FC = () => (
```

Comment:
Move this to a separate global component inside `packages/ui` so we can make sure it as a standard